File: 3/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        hash_set = set()
        res = 0
        i = j = 0
        n = len(s)
        while j < n:
            c = s[j]
            while i < j and c in hash_set:
                hash_set.discard(s[i])
                i += 1
            hash_set.add(c)
            j += 1
            res = max(res, j - i)
        return res

File: 3/Solution.java
import java.util.HashSet;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int res = 1;
        int n = s.length();
        int i = 0, j = 0;
        HashSet<Character> hashSet = new HashSet<>();
        while (j < n) {
            char c = s.charAt(j);
            while (hashSet.contains(c)) {
                hashSet.remove(s.charAt(i));
                i++;
            }
            hashSet.add(c);
            j++;
            res = Math.max(res, j - i);
        }
        return res;
    }
}